darts/utils/data/encoders.py

```diff
@@ -0,0 +1,891 @@
+        """
+        super(CallableIndexEncoder, self)._encode(index)
+
+        return TimeSeries.from_times_and_values(times=index,
+                                                values=self.attribute(index),
+                                                columns=['custom']).astype(np.dtype(self.dtype))
+
+    @property
+    def accept_transformer(self) -> List[bool]:
+        """CallableIndexEncoder accepts transformations."""
+        return [True]
+
+
+class PastCallableIndexEncoder(CallableIndexEncoder):
+    """IntegerIndexEncoder: Adds integer index value (position) for past covariates derived from the underlying
+    TimeSeries' time index.
+    """
+
+    def __init__(self, input_chunk_length: int, output_chunk_length: int, attribute: Union[str, Callable]):
+        """
+        Parameters
+        ----------
+        input_chunk_length
+            The length of the emitted past series.
+        output_chunk_length
+            The length of the emitted future series.
+        attribute
+            A callable that takes an index `index` of type `(pd.DatetimeIndex, pd.Int64Index, pd.RangeIndex)` as input
+            and returns a np.ndarray of shape `(len(index),)`.
+            An example for a correct `attribute` for `ìndex` of type pd.DatetimeIndex:
+                `attribute = lambda index: (index.year - 1950) / 50}`
+            And for pd.Int64Index or pd.RangeIndex:
+                `attribute = lambda index: (index - 1950) / 50}`
+        """
+        super(PastCallableIndexEncoder, self).__init__(
+            index_generator=PastCovariateIndexGenerator(input_chunk_length, output_chunk_length),
+            attribute=attribute
+        )
+
+
+class FutureCallableIndexEncoder(CallableIndexEncoder):
+    """IntegerIndexEncoder: Adds integer index value (position) for future covariates derived from the underlying
+    TimeSeries' time index.
+    """
+
+    def __init__(self, input_chunk_length: int, output_chunk_length: int, attribute: Union[str, Callable]):
+        """
+        Parameters
+        ----------
+        input_chunk_length
+            The length of the emitted past series.
+        output_chunk_length
+            The length of the emitted future series.
+        attribute
+            A callable that takes an index `index` of type `(pd.DatetimeIndex, pd.Int64Index, pd.RangeIndex)` as input
+            and returns a np.ndarray of shape `(len(index),)`.
+            An example for a correct `attribute` for `ìndex` of type pd.DatetimeIndex:
+                `attribute = lambda index: (index.year - 1950) / 50]}`
+            And for pd.Int64Index or pd.RangeIndex:
+                `attribute = lambda index: (index - 1950) / 50]}`
+        """
+
+        super(FutureCallableIndexEncoder, self).__init__(
+            index_generator=FutureCovariateIndexGenerator(input_chunk_length, output_chunk_length),
+            attribute=attribute
+        )
+
+
+class SequenceEncoder(Encoder):
```

Comment:
The name `SequenceEncoder` might be slightly misleading as it could be understood as an encoder for a sequence. How about e.g. `EncoderSequence` ?

Reply:
What do you think about `SequentialEncoder`? Would be consistent with the style I used for the other classes  `{X}Encoder` like `SingleEncoder`

Reply:
Sounds good!